src/c++/library/grpc_client.cc

```diff
@@ -135,12 +135,15 @@ GetStub(
       grpc::CreateCustomChannel(url, credentials, arguments);
   std::shared_ptr<inference::GRPCInferenceService::Stub> stub =
       inference::GRPCInferenceService::NewStub(channel);
-  // Replace if channel / stub have been in the map
-  if (channel_itr != grpc_channel_stub_map_.end()) {
-    channel_itr->second = std::make_tuple(1, channel, stub);
-  } else {
-    grpc_channel_stub_map_.insert(
-        std::make_pair(url, std::make_tuple(1, channel, stub)));
+
+  if (use_cached_channel) {
```

Comment:
I wouldn't expect testing changes to change production code. Is this needed?

Reply:
This change is coming from here: https://github.com/triton-inference-server/client/pull/387
I do not know what are process is for PRs without testing so made a branch from the external PR, added testing and made sure the commit shows the original contributors github handle.